compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/FirDeclarationsResolveTransformer.kt

```diff
@@ -496,8 +498,12 @@ open class FirDeclarationsResolveTransformer(transformer: FirBodyResolveTransfor
 
         val body = result.body
         if (result.returnTypeRef is FirImplicitTypeRef) {
+            val simpleFunction = function as? FirSimpleFunction
```

Comment:
Should we do the similar for property with initializer?

Reply:
Oops, the place I added covered functions _and_ accessors, so I thought that would be enough. Didn't think about that, added. Thanks!